miio/integrations/roborock/vacuum/vacuum.py

```diff
@@ -515,20 +522,68 @@ def enable_lab_mode(self, enable):
     @command()
     def clean_history(self) -> CleaningSummary:
         """Return generic cleaning history."""
-        return CleaningSummary(self.send("get_clean_summary"))
+        self._clean_history = CleaningSummary(self.send("get_clean_summary"))
+        return self._clean_history
 
     @command()
-    def last_clean_details(self) -> Optional[CleaningDetails]:
+    def last_clean_details(self, skip_cache=False) -> Optional[CleaningDetails]:
         """Return details from the last cleaning.
 
         Returns None if there has been no cleanups.
         """
-        history = self.clean_history()
-        if not history.ids:
+        if self._clean_history is None or skip_cache:
+            self.clean_history()
+        assert isinstance(self._clean_history, CleaningSummary)  # nosec assert_used
+        if not self._clean_history.ids:
             return None
 
-        last_clean_id = history.ids.pop(0)
-        return self.clean_details(last_clean_id)
+        last_clean_id = self._clean_history.ids[0]
+        if last_clean_id == self._searched_clean_id:
+            return self._last_clean_details
+
+        self._last_clean_details = self.clean_details(last_clean_id)
+        return self._last_clean_details
+
+    @command()
+    def last_clean_all_floor(self, skip_cache=False) -> FloorCleanDetails:
+        """Return details from the last cleaning and for each floor.
+
+        Returns None if there has been no cleanups for that floor.
+        """
+        if self._clean_history is None or skip_cache:
+            self.clean_history()
+        assert isinstance(self._clean_history, CleaningSummary)  # nosec assert_used
```

Comment:
Let's remove the caching from this PR and concentrate on getting the multi map handling implemented.